operators/pkg/controller/elasticsearch/sset/getter.go

```diff
@@ -15,3 +17,20 @@ func Replicas(statefulSet appsv1.StatefulSet) int32 {
 	}
 	return 0
 }
+
+// UpdatePartition returns the updateStrategy.Partition index, or falls back to the number of replicas if not set.
+func UpdatePartition(statefulSet appsv1.StatefulSet) int32 {
+	rollingUpdate := statefulSet.Spec.UpdateStrategy.RollingUpdate
+	if rollingUpdate != nil && rollingUpdate.Partition != nil {
+		return *rollingUpdate.Partition
+	}
+	if statefulSet.Spec.Replicas != nil {
+		return *statefulSet.Spec.Replicas
+	}
+	return 0
```

Comment:
Should this be -1? I assume partitions are numbered from 0 so right now there is no result difference between error path and happy path - could that be an issue?

Reply:
I think this is a rather sane default. We always explicitly set replicas and update partition, but if a user happens to set it back to nil explicitly (not sure why they would do that), we consider all pods can be upgraded.

Returning -1 here could have other unintended consequences I think (not sure what happens if we explicitly update the partition to -1 for example).

Reply:
-1 would skip entire loop, I'm not sure where 0 with error would end up. If you feel it's safe and clear then I'm fine with 0 though:)